internal/cri/server/stats.go

```diff
@@ -0,0 +1,200 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	wstats "github.com/Microsoft/hcsshim/cmd/containerd-shim-runhcs-v1/stats"
+	cg1 "github.com/containerd/cgroups/v3/cgroup1/stats"
+	cg2 "github.com/containerd/cgroups/v3/cgroup2/stats"
+	"github.com/containerd/containerd/v2/api/services/tasks/v1"
+	ctrdutil "github.com/containerd/containerd/v2/internal/cri/util"
+	"github.com/containerd/containerd/v2/protobuf"
+
+	"github.com/containerd/containerd/v2/api/types"
+	"github.com/containerd/containerd/v2/internal/cri/store/stats"
+	"github.com/containerd/log"
+)
+
+// MetricMonitor is used to monitor stat.
+type metricMonitor struct {
+	c                *criService
+	collectionPeriod time.Duration
+}
+
+// newMetricMonitor creates a new monitor which will collect stats for containers and sandboxes
+// every 10 seconds and calculate the NanoCores Usage
+func newMetricMonitor(c *criService) *metricMonitor {
+	return &metricMonitor{
+		c:                c,
+		collectionPeriod: 10 * time.Second,
```

Comment:
That's an arbitrary value (inherited from another scenario). Should this become configurable in the long term?